src/test/java/jewellery/inventory/integration/ResourceInOrganizationCrudIntegrationTest.java

```diff
@@ -0,0 +1,342 @@
+
+  @Test
+  void removeResourceFromDatabaseWhenResourceQuantityIsZero() {
+    OrganizationResponseDto organizationResponseDto = createOrganization();
+    ResourceResponseDto resourceResponse = createResourceResponse();
+
+    ResourceInOrganizationRequestDto request =
+        ResourceInOrganizationTestHelper.createResourceInOrganizationRequestDto(
+            organizationResponseDto.getId(),
+            resourceResponse.getId(),
+            RESOURCE_QUANTITY_TO_REMOVE,
+            RESOURCE_PRICE);
+
+    sendResourceToOrganization(request);
+    ResponseEntity<ResourcesInOrganizationResponseDto> resourceResponseBeforeDeletingQuantity =
+        getAllResourcesByOrganizationId(organizationResponseDto);
+    assertEquals(
+        1, resourceResponseBeforeDeletingQuantity.getBody().getResourcesAndQuantities().size());
+
+    sendDeleteOperation(getDeleteResourceUrl(organizationResponseDto, resourceResponse));
+    ResponseEntity<ResourcesInOrganizationResponseDto> resourceResponseAfterDeletingQuantity =
+        getAllResourcesByOrganizationId(organizationResponseDto);
+
+    assertEquals(
+        0, resourceResponseAfterDeletingQuantity.getBody().getResourcesAndQuantities().size());
+  }
+
+  @Test
+  void removeResourceQuantityShouldThrowWhenOrganizationNotFound() {
+    OrganizationResponseDto organizationResponseDto = createOrganization();
+    ResourceResponseDto resourceResponse = createResourceResponse();
+
+    organizationResponseDto.setId(UUID.randomUUID());
+
+    ResponseEntity<ResourcesInOrganizationResponseDto> response =
+        sendDeleteOperation(getDeleteResourceUrl(organizationResponseDto, resourceResponse));
+    assertEquals(HttpStatus.NOT_FOUND, response.getStatusCode());
+  }
+
+  @Test
+  void removeResourceQuantityShouldThrowWhenResourceNotFound() {
+    OrganizationResponseDto organizationResponseDto = createOrganization();
+    ResourceResponseDto resourceResponse = createResourceResponse();
+
+    resourceResponse.setId(UUID.randomUUID());
+
+    ResponseEntity<ResourcesInOrganizationResponseDto> response =
+        sendDeleteOperation(getDeleteResourceUrl(organizationResponseDto, resourceResponse));
+    assertEquals(HttpStatus.NOT_FOUND, response.getStatusCode());
+  }
+
+  @Test
+  void removeResourceQuantityShouldThrowWhenInsufficientQuantity() {
+    OrganizationResponseDto organizationResponseDto = createOrganization();
+    ResourceResponseDto resourceResponse = createResourceResponse();
+
+    ResourceInOrganizationRequestDto request =
+        ResourceInOrganizationTestHelper.createResourceInOrganizationRequestDto(
+            organizationResponseDto.getId(),
+            resourceResponse.getId(),
+            RESOURCE_QUANTITY_TO_REMOVE.subtract(getBigDecimal("1")),
+            RESOURCE_PRICE);
+
+    sendResourceToOrganization(request);
+
+    ResponseEntity<ResourcesInOrganizationResponseDto> response =
+        sendDeleteOperation(getDeleteResourceUrl(organizationResponseDto, resourceResponse));
```

Comment:
Have this separated both above and below it so that it is separated as "action" and the the last separation is assertions

Reply:
Done!